### PR TITLE
Retry `apt-get update` after 10s in dockerfiles

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR attempts to fix the flakiness of dockerfile builds in CI by retrying `apt-get update` up to 4 times, each after a 10s timeout. Hopefully this will eliminate the majority of exit code `100`s (at the associated manual retries) seen on CI.

Checklist:

- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Review by lk86 — CHANGES_REQUESTED** [review]
I think 95% if not more of the apt-get update failures are our debian repository failing, so we shouldn't need this mechanism in all of these places. Just when we are installing mina-daemon, mina-archive, etc. And I would still rather try and improve the repository itself in some way but this is a workable stop-gap.

**Comment by mrmr1993** [comment]
Refined it to only retry after `packages.o1test.net` have been added to the sources list.

File: dockerfiles/stages/4-production [hunk 1]
```diff
@@ -16,6 +16,9 @@ ENV DEBIAN_FRONTEND noninteractive
 
 # --- Dependencies across many platforms
 RUN apt-get update -y \
+  || sleep 10s && apt-get update -y \
```
Comment (lk86) [inline dockerfiles/stages/4-production:19]: These (both uses in this file) do not use o1test.net either

**Review by lk86 — APPROVED** [review]

File: dockerfiles/stages/4-production (hunk 1 above)
Comment (mrmr1993) [reply dockerfiles/stages/4-production]: My bad, I forgot to check in that file. Thanks!

**Pull request merged**